pkg/repl/client_test.go

```diff
@@ -758,3 +760,110 @@ func TestAllChangesFlushed(t *testing.T) {
 	subQueue.HasChanged([]any{3}, nil, false)
 	assert.False(t, client.AllChangesFlushed(), "Should not be flushed with items in queue")
 }
+
+// TestMaxRecreateAttempts validates the maxRecreateAttempts variable and verifies
+// the counter logic works correctly (increment and reset).
+func TestMaxRecreateAttempts(t *testing.T) {
+	// Verify the variable exists and has the correct default value
+	assert.Equal(t, maxRecreateAttemptsDefault, maxRecreateAttempts, "maxRecreateAttempts should default to 10")
+
+	// Test that we can modify it for testing
+	originalValue := maxRecreateAttempts
+	defer func() { maxRecreateAttempts = originalValue }()
+
+	maxRecreateAttempts = 5
+	assert.Equal(t, 5, maxRecreateAttempts)
+
+	maxRecreateAttempts = originalValue
+	assert.Equal(t, maxRecreateAttemptsDefault, maxRecreateAttempts)
+}
+
+// TestMaxRecreateAttemptsPanic tests that the panic actually occurs after max attempts.
+// This uses a subprocess pattern to test the panic behavior.
+func TestMaxRecreateAttemptsPanic(t *testing.T) {
+	if os.Getenv("TEST_PANIC_SUBPROCESS") == "1" {
+		// This is the subprocess that should panic
+		testMaxRecreateAttemptsPanicSubprocess(t)
+		return
+	}
+
+	// Run the test in a subprocess
+	cmd := exec.Command(os.Args[0], "-test.run=TestMaxRecreateAttemptsPanic")
+	cmd.Env = append(os.Environ(), "TEST_PANIC_SUBPROCESS=1")
+	output, err := cmd.CombinedOutput()
+
+	outputStr := string(output)
+	t.Logf("Subprocess output:\n%s", outputStr)
+
+	// We expect the subprocess to exit with non-zero (panic or crash)
+	if err == nil {
+		t.Fatal("Expected subprocess to panic or crash, but it exited successfully")
+	}
+
+	if !strings.Contains(outputStr, "consecutive errors") {
+		t.Errorf("Expected to see consecutive errors. Output:\n%s", outputStr)
+	}
+	if !strings.Contains(outputStr, "Failed to recreate streamer") {
+		t.Errorf("Expected to see recreation attempt. Output:\n%s", outputStr)
+	}
+
+	// If we DID get the max attempts panic message, verify it's correct
+	if strings.Contains(outputStr, "failed to recreate binlog streamer after") {
+		if !strings.Contains(outputStr, "giving up") {
+			t.Errorf("Panic message should contain 'giving up'. Output:\n%s", outputStr)
+		}
```

Comment:
The technique here to run a sub-process is very clever :-) But I think it's better to either add a recover for panic, or see if you can get this to work correctly with context canceling. If you pursue context canceling, we need to make sure it works end-to-end because the migration caller should also detect that the replClient has failed and bail.